src/read_write/codec.rs

```diff
@@ -140,3 +140,67 @@ where
         .unwrap()
         .mul_add(edge_length, min)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn plain_scalar() {
+        let value = 41.33333;
+        let min = 40.0;
+        let edge_length = 2.0;
+
+        let value_f32 = _encode::<f32>(value, min, edge_length);
+        let value_f32_decoded = decode(value_f32, min, edge_length);
+        assert!(
```

Comment:
nit: how about we use [approx](https://docs.rs/approx/0.3.2/approx/) crate here and possibly other places?

Reply:
Probably a good idea, but I'd do that separately since I don't want to do it in this one place only and otherwise this PR becomes an "Introduce approx" PR. Leaving a comment in the code.